### PR TITLE
ACT24 automation + some code clean up

### PR DESCRIPTION
**Pull request by matakevin**
This PR contains some progress on the ACT24 application automation and some clean up for pageInitializer and cedcd classes

### CODE REVIEW

File: src/test/java/AnalysisTools/ACT24/Steps/ACT24ResearcherPortalSteps.java
Comment (Mariachaudhry) [inline src/test/java/AnalysisTools/ACT24/Steps/ACT24ResearcherPortalSteps.java:1]: Please add line between 19, 20 and 40,41

**Review by Mariachaudhry — APPROVED** [review]

File: src/test/java/AnalysisTools/ACT24/Features/ParticipantPortal.feature [hunk 1]
```diff
@@ -1,2 +1,14 @@
...
\ No newline at end of file
+
+  Background: Participant is logged in
+    Given the participant on the ACT24 participant
+
+  @Smoke @matakevin
+  Scenario: The Participant splash screen is display
+    Then the splash screen is displayed
+
+   @matakevin
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/ACT24/Features/ParticipantPortal.feature:10]: Please add @Regression or @Smoke tags plus the Jira story#

File: src/test/java/AnalysisTools/ACT24/Features/ParticipantPortal.feature [hunk 2]
```diff
@@ -1,2 +1,14 @@
 Feature: These are the features on the Participant Portal
-  Scenario:
\ No newline at end of file
+
+  Background: Participant is logged in
+    Given the participant on the ACT24 participant
+
+  @Smoke @matakevin
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/ACT24/Features/ParticipantPortal.feature:6]: Please add @regression

File: src/test/java/AnalysisTools/ACT24/Features/ResearcherPortal.feature [hunk 3]
```diff
@@ -1,2 +1,16 @@
 Feature: These are the features on the Researcher Portal
-  Scenario:
\ No newline at end of file
+
+  @Smoke @matakevin
+  Scenario: The Researcher can log in
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/ACT24/Features/ResearcherPortal.feature:4]: Please add @regression and the Jira story#

File: src/test/java/AnalysisTools/ACT24/Features/ResearcherPortal.feature [hunk 4]
```diff
@@ -1,2 +1,16 @@
...
+    Given the user is on the ACT24 Researcher site
+    When the user enters "Kevin.MataRodriguez@nih.gov" as email and "ACT24Test!" as password and clicks Login
+    Then the Researcher is logged in
+
+  @Smoke @matakevin
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/ACT24/Features/ResearcherPortal.feature:9]: Please add @regression and the Jira story#

File: src/test/java/AnalysisTools/ACT24/Steps/ACT24ParticipantPortalSteps.java [hunk 5]
```diff
@@ -1,6 +1,39 @@
...
+import org.testng.Assert;
 
 public class ACT24ParticipantPortalSteps extends PageInitializer {
-}
+
+    @Given("the participant on the ACT24 participant")
+    public void the_participant_on_the_act24_participant() {
+        WebDriverUtils.webDriver.get(EnvUtils.getApplicationUrl("ACT24Participant"));
+        MiscUtils.sleep(2000);
+        Assert.assertTrue(act24ParticipantPortalPage.act24WelcomeText.isDisplayed());
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/ACT24/Steps/ACT24ParticipantPortalSteps.java:20]: Shouldn't these be implemented in Steps Implementation class? Per Diego, it is fine as long as it is one line of code.

File: src/test/java/AnalysisTools/ACT24/Steps/ACT24ParticipantPortalSteps.java [hunk 6]
```diff
@@ -1,6 +1,39 @@
...
+
+    @Then("the splash screen is displayed")
+    public void the_splash_screen_is_displayed() {
+        Assert.assertTrue(act24ParticipantPortalPage.splashScreenText.isDisplayed());
+    }
+
+    @When("the user adds wake up time on splash screen")
+    public void the_user_adds_wake_up_time_on_splash_screen() {
+        act24ParticipantStepImpl.addWakeTimeOnSplashScreen();
+    }
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/ACT24/Steps/ACT24ParticipantPortalSteps.java:31]: Please add a space after line 31

File: src/test/java/AnalysisTools/ACT24/Steps/ACT24ParticipantPortalSteps.java [hunk 7]
```diff
@@ -1,6 +1,39 @@
...
+    @When("the user adds a Sleeping or in Bed entry")
+    public void the_user_adds_a_sleeping_or_in_bed_entry() {
+        CommonUtils.clickOnElement(act24ParticipantPortalPage.okOnWelcomeBackPopup);
+    }
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/ACT24/Steps/ACT24ParticipantPortalSteps.java:35]: Please add a space after line 35

File: src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24ProgressionTest.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
+package AnalysisTools.AnalysisToolsRunners;
+
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
+
+@CucumberOptions(plugin = {"html:target/html-reports/cucumber-default-report.html", "json:target/cucumber.json",
+        "junit:target/cucumber.xml", "rerun:target/failed.txt",
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24ProgressionTest.java:7]: Please remove "junit:target/cucumber.xml" from you runner class as we no longer use jUnit.

File: src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24ProgressionTest.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+        "pretty", "com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"},
+        features = "src/test/java/AnalysisTools/ACT24/Features",
+        glue = "AnalysisTools.ACT24.Steps",
+        tags = "@Progression",
+        dryRun = true
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24ProgressionTest.java:12]: Please set dryRun to false

File: src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24RegressionTest.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
+package AnalysisTools.AnalysisToolsRunners;
+
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
+
+@CucumberOptions(plugin = {"html:target/html-reports/cucumber-default-report.html", "json:target/cucumber.json",
+        "junit:target/cucumber.xml", "rerun:target/failed.txt",
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24RegressionTest.java:7]: Please remove "junit:target/cucumber.xml" from you runner class as we no longer use jUnit.

File: src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24SmokeTest.java [hunk 11]
```diff
@@ -0,0 +1,23 @@
+package AnalysisTools.AnalysisToolsRunners;
+
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
+
+@CucumberOptions(plugin = {"html:target/html-reports/cucumber-default-report.html", "json:target/cucumber.json",
+        "junit:target/cucumber.xml", "rerun:target/failed.txt",
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24SmokeTest.java:7]: Please remove "junit:target/cucumber.xml" from you runner class as we no longer use jUnit.

File: src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24RegressionTest.java [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+/**
+ * This class is annotated with @RunWith(Cucumber.class) and it will run
+ * cucumber feature(s).
+ *
+ * @author sohilz2
+ */
+public class RunACT24RegressionTest extends AbstractTestNGCucumberTests{
+
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24RegressionTest.java:22]: Please remove the empty line 22

File: src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24SmokeTest.java [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+/**
+ * This class is annotated with @RunWith(Cucumber.class) and it will run
+ * cucumber feature(s).
+ *
+ * @author sohilz2
+ */
+public class RunACT24SmokeTest extends AbstractTestNGCucumberTests{
+
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24SmokeTest.java:22]: Please remove the empty line 22

File: src/test/java/AnalysisTools/AnalysisToolsRunners/RunJPSurvSmokeTest.java [hunk 14]
```diff
@@ -14,14 +14,8 @@
 		, features="src/test/java/AnalysisTools/JPSurv/Features"
 		, glue="AnalysisTools.JPSurv.Steps"
 		, tags="@Smoke"
-		, dryRun = false
+		, dryRun=false
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/AnalysisToolsRunners/RunJPSurvSmokeTest.java:17]: Please remove line 12, "junit:target/cucumber.xml" from you runner class as we no longer use jUnit.

**Review by hamidnesar — CHANGES_REQUESTED** [review]
Please review the comments. Thank you!

File: src/test/java/AnalysisTools/ACT24/Features/ParticipantPortal.feature (hunk 2 above)
Comment (matakevin) [reply src/test/java/AnalysisTools/ACT24/Features/ParticipantPortal.feature:6]: I'll add the regression tags on the runners

File: src/test/java/AnalysisTools/ACT24/Features/ResearcherPortal.feature (hunk 3 above)
Comment (matakevin) [reply src/test/java/AnalysisTools/ACT24/Features/ResearcherPortal.feature:4]: Analysis Tools doesn't use user stories to automate scenarios.  The dev team asks us to automate certain scenarios (regardless if there's a requirement for it or not.). We hope to change this process soon

File: src/test/java/AnalysisTools/ACT24/Steps/ACT24ParticipantPortalSteps.java (hunk 5 above)
Comment (matakevin) [reply src/test/java/AnalysisTools/ACT24/Steps/ACT24ParticipantPortalSteps.java:20]: Lines 19 and 20 aren't necessary but I wanted to make sure the welcome page was displayed.  I'll remove these lines

File: src/test/java/AnalysisTools/ACT24/Steps/ACT24ResearcherPortalSteps.java [hunk 15]
```diff
@@ -1,7 +1,52 @@
...
+import org.testng.Assert;
 
 public class ACT24ResearcherPortalSteps extends PageInitializer {
 
-}
+    @Given("the user is on the ACT24 Researcher site")
+    public void the_user_is_on_the_act24_researcher_site() {
+        WebDriverUtils.webDriver.get(EnvUtils.getApplicationUrl("ACT24Researcher"));
+        MiscUtils.sleep(2000);
+        Assert.assertTrue(act24ResearcherPortalPage.researcherText.isDisplayed());
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/ACT24/Steps/ACT24ResearcherPortalSteps.java:18]: Are lines 17 and 18 necessary?

File: src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24ProgressionTest.java [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
+
+@CucumberOptions(plugin = {"html:target/html-reports/cucumber-default-report.html", "json:target/cucumber.json",
+        "rerun:target/failed.txt",
+        "pretty", "com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"},
+        features = "src/test/java/AnalysisTools/ACT24/Features",
+        glue = "AnalysisTools.ACT24.Steps",
+        tags = "@Progression",
+        dryRun = true
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/AnalysisToolsRunners/RunACT24ProgressionTest.java:12]: Please set dryRun to false

File: src/test/java/AnalysisTools/ACT24/Steps/ACT24ParticipantPortalSteps.java [hunk 17]
```diff
@@ -1,6 +1,37 @@
...
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/ACT24/Steps/ACT24ParticipantPortalSteps.java:33]: Please add a space after line 33

File: src/test/java/AnalysisTools/ACT24/Steps/ACT24ParticipantPortalSteps.java [hunk 18]
```diff
@@ -1,6 +1,37 @@
...
+
+    @Then("the splash screen is displayed")
+    public void the_splash_screen_is_displayed() {
+        Assert.assertTrue(act24ParticipantPortalPage.splashScreenText.isDisplayed());
+    }
+
+    @When("the user adds wake up time on splash screen")
+    public void the_user_adds_wake_up_time_on_splash_screen() {
+        act24ParticipantStepImpl.addWakeTimeOnSplashScreen();
+    }
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/ACT24/Steps/ACT24ParticipantPortalSteps.java:29]: Please add a space after line 29

**Review by hamidnesar — CHANGES_REQUESTED** [review]
Please review the comments.

File: src/test/java/AnalysisTools/ACT24/Steps/ACT24ResearcherPortalSteps.java [hunk 19]
```diff
@@ -1,7 +1,50 @@
...
+        act24ResearcherStepImpl.researcherLogin(email, password);
+        MiscUtils.sleep(2000);
+        act24ResearcherStepImpl.researcherLogin(email, password);
+    }
+
+    @When("the user creates a new study {string}, {string}, {string}, {string}, {string}, {string}, {string}")
+    public void the_user_creates_a_new_study(String studyName, String studyDescription, String studyAbbreviation, String expectedParticipants, String recallPerParticipant, String studyStartDate, String studyEndDate) {
+        MiscUtils.sleep(3000);
+        act24ResearcherStepImpl.createNewStudy(studyName, studyDescription, studyAbbreviation, expectedParticipants, recallPerParticipant, studyStartDate, studyEndDate);
+        //NEED TO CHANGE STUDY ABBREVIATION (THERE CAN'T BE DUPLICATE NAMES)
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/ACT24/Steps/ACT24ResearcherPortalSteps.java:38]: Is the comment on line 38 necessary?

**Review by hamidnesar — APPROVED** [review]
Looks good!
Is the comment on line 38 necessary in ACT24ResearcherPortalSteps.java?

**Review by alenaminsk — APPROVED** [review]

**Pull request merged**